### PR TITLE
Fix Debug Build On Windows

### PR DESCRIPTION
**Pull request by lara-hdr**
compile files in torch/csrc with /MDd runtime library option for debug build on Windows

### CODE REVIEW

File: torch/CMakeLists.txt [hunk 1]
```diff
@@ -205,24 +205,45 @@ target_compile_definitions(torch PRIVATE _THP_CORE)
...
+		/wd4244
+		/wd4190
+		/wd4101
+		/wd4996
+		/wd4275
+		/bigobj
+		)
+	else()
+	  target_compile_options(torch PRIVATE
+		/MD
```
Comment (peterjc123) [inline torch/CMakeLists.txt]: Is it possible to just substitute `/MD` with a variable? Copy and paste code is not good under this condition.

**Review by peterjc123 — CHANGES_REQUESTED** [review]
See inline comments

File: torch/CMakeLists.txt (hunk 1 above)
Comment (ezyang) [reply torch/CMakeLists.txt]: Yeah, I'm not so hot on copy pasting all the warnings.

File: torch/CMakeLists.txt [hunk 2]
```diff
@@ -205,30 +205,30 @@ target_compile_definitions(torch PRIVATE _THP_CORE)
...
-	  /wd4838
-	  /wd4305
-	  /wd4244
-	  /wd4190
-	  /wd4101
-	  /wd4996
-	  /wd4275
-	  /bigobj
-	  )
+  if (${CMAKE_BUILD_TYPE} MATCHES "Debug")
```
Comment (peterjc123) [inline torch/CMakeLists.txt]: Why do you replace tabs with spaces? The remaining part of the code uses tabs. Let's keep it.

Comment (lara-hdr) [reply torch/CMakeLists.txt]: the remaining code uses spaces

Comment (ezyang) [reply torch/CMakeLists.txt]: I'm quite happy to see tabs go :)

**Review by peterjc123 — APPROVED** [review]
LGTM.

**Comment by peterjc123** [comment]
@ezyang @soumith Can we please merge this PR?

File: torch/CMakeLists.txt [hunk 3]
```diff
@@ -206,8 +206,14 @@ target_compile_definitions(torch PRIVATE _THP_CORE)
 
 # until they can be unified, keep these lists synced with setup.py
 if(MSVC)
+  if (${CMAKE_BUILD_TYPE} MATCHES "Debug")
+    set (MSVC_RUNTIME_LIBRARY_FLAG "/MDd")
+  else()
+    set (MSVC_RUNTIME_LIBRARY_FLAG "/MD")
+  endif()
+
```
Comment (peterjc123) [inline torch/CMakeLists.txt]: nit: remove spaces in this line.

**Pull request closed without merging**